pyuvdata/uvcal/initializers.py

```diff
@@ -114,6 +124,10 @@ def new_uvcal(
         gain_array). By default, this function creates a metadata-only object. You can
         pass in data arrays directly to the constructor if you want to create a
         fully-populated object.
+    data : dict of array_like, optional
+        Dictionary containing optional data arrays. Possible keys are:
+        'gain_array', 'delay_array', 'quality_array', 'flag_array',
+        and 'total_quality_array'.
```

Comment:
What happens if only some of the keys are provided? Some of these are optional but I think some are required and we should at least tell users that if any are provided that they might get an error in the check if they don't provide at least the minimal set.

Reply:
I've added more detail to this docstring, making it clear that a subset can be provided and the others will then be set to default "empty" values.